### DISCUSSION TITLE
rendering with .py to create a dashboard using streamlit

### DISCUSSION THREAD

**Discussion by Kkhurana007** [quarto-dev/quarto-cli#9872] [Q&A]
### Description

when i try to run the dashboard with Streamlit, it works fine. here is the code: https://github.com/Kkhurana007/dashboarding/blob/main/dashboard.py

But, I want to render it to my blog using Quarto. So, I converted `.py` to `.ipynb` file: https://github.com/Kkhurana007/dashboarding/blob/main/dashboard.ipynb  

As a result, Jupyter notebook gets displayed and not the interactive dashboard. Also, when i run it in the locally, I get DeltaGenerato() 
![image](https://github.com/quarto-dev/quarto-cli/assets/91566244/0e91e5d8-90a7-456b-aa9b-de72b7564a1c)

Furthermore, I try to run the `ipykernel_launcher.py` argument. details copied below:  

import sys
from pathlib import Path

if __name__ == "__main__":
    # Remove the CWD from sys.path while we load stuff.
    # This is added back by InteractiveShellApp.init_path()
    if sys.path[0] == "" or Path(sys.path[0]) == Path.cwd():
        del sys.path[0]

    from ipykernel import kernelapp as app

    app.launch_new_instance()

I get directed here: 

![image](https://github.com/quarto-dev/quarto-cli/assets/91566244/7473860d-db85-4a35-9700-b31d64d2f9b2)

If anyone can help to create an interactive dashboard from `.ipynb`? 

Sincerely,
Kunal Khurana

**Comment by mcanouil** [comment]
You have an obvious error: `format: dashbord`.

Also I suggest to read the dashboard guide at <https://quarto.org/docs/dashboards/>.

It's not simply about using a Jupyter Notebook with `format: dashboard` in the metadata to get a "dashboard".
This being said, with the proper format name:

<img width="367" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/b999cf8a-df42-466c-a2c4-8fef40260b6f">

Edit: your metadata are set in a Python code cell. It needs to be markdown or as above the default will be `format: html`.

I suggest you start from scratch and add elements following the guide or using a pre-existing Jupyter based dashboard that you modify.

**Reply by Kkhurana007** [reply]
Thanks for your response @mcanouil 
I believe I got a bit closer to what i want. 
![image](https://github.com/quarto-dev/quarto-cli/assets/91566244/233341fc-f6ee-468a-be5d-7e0e9ebe3324)


Do, you have any suggestion to fix Delta Generator() ? 
![image](https://github.com/quarto-dev/quarto-cli/assets/91566244/4d40df39-85ec-4572-90a0-3a559f1a5ee2)

Sincerely, 
Kunal Khurana

**Comment by cderv** [comment]
> when i try to run the dashboard with Streamlit, it works fine. (...)
> But, I want to render it to my blog using Quarto. So, I converted .py to .ipynb file

I believe some clarification may help here. What are you trying to do exactly ? 

Are you trying to show a streamlit dashboard in a Quarto website ? 
Or are you trying to convert a streamlit dashboard to a Quarto Dashboard (https://quarto.org/docs/dashboards/)

The above answer assumes the latter, but I am wondering if you are not trying to do the former. 🤔

**Reply by Kkhurana007** [reply]
Thanks for your response. I'm trying to attempt the former. Trying to show a streamlit dashboard on a quarto website.

**Comment by mcanouil** [comment]
@Kkhurana007 Could you confirm something: does streamlit require a running Python environment to work?

it does seem so.

<table>
<tr><th>Quarto document</th><th>HTML</th></tr>
<tr><td>

````qmd
---
title: Getting the min and max date
format: html
jupyter: python3
---

```{python}
import streamlit as st
import pandas as pd
import numpy as np

st.title('Uber pickups in NYC')
```

````

</td><td>

<img width="614" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/d16cce9d-b922-43d4-95f3-24926e2fbb69">

</td></tr>
</table>

**Reply by Kkhurana007** [reply]
@mcanouil I believe, yes! When you select particular items the graphs get customised accordingly!

**Reply by mcanouil** [reply]
Then I don't think you can use streamlit.
Note that you would have the same issue in Jupyter Notebooks alone.

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/c97e5510-6363-45b3-8432-0776052e6511">

**Reply by Kkhurana007** [reply]
@mcanouil Ah, I see. do you think it is wise to seek help from Jupyter community for using streamlit as extension for live dashboards? Perhaps they might have something to add!

**Reply by mcanouil** [reply]
The thing is I'm not sure about Quarto ability to run jointly with a Python session. (cc @cscheid)
Alternatively, there is Shiny for Python (https://shiny.posit.co/py/).

**Reply by Kkhurana007** [reply]
> The thing is I'm not sure about Quarto ability to run jointly with a Python session. (cc @cscheid) Alternatively, there is Shiny for Python (https://shiny.posit.co/py/).

Awesome, I'll look into that! thanks a lot for your feedback!!!
